### ISSUE TITLE
Expending the player's stats to the whole game

### ISSUE THREAD

**Issue by AnddyAnddy** [haxball/haxball-issues#908]
Well, I believe this already has been asked.

In almost every games, our stats through time are saved regardless the session we are in, example: number of wins, number of goals scored, time spent in the game, number of assists... Basic haxball stuff at all.

Maybe it would need a big update to do this, but I believe it's doable, it would mean that you have to automatically add the basic stats counts to ALL rooms but you can find the best snippets from all of them from your community (and even from yourself, since I assume you just coded everything you wanted).

Some scripts allows this kind of idea through several rooms or even website but we never have these features whatever the room is.

**Comment by Wazarr94** [comment]
Since rooms have settings that can very much vary, I believe it's best we stay as we currently are.

By nature, HaxBall rooms' structure is very loose. Even if basro was eager to implement global stats, I don't see how he can make it relevant.
What stops me from creating a room in which I can easily farm stats ? Anyone could use a simple script that gives you 300 goals, 200 wins and 300 assists in 10 minutes, using the Headless API.

But of course, if official HaxBall rooms were introduced, then we could be seriously looking into it.

**Comment by AnddyAnddy** [comment]
1) if someone farms stats, it's his business, I assume there won't be ranking

2) best scripts (including mines) prevents a player to farm by a lot of ways

In fps as an example there are private and public rooms, important stats doesn't count in privates. Since in haxball, privates room are more used than publics, this split can't work like that, but you see the idea behind this

**Comment by Wazarr94** [comment]
The other point in my answer is how different HaxBall rooms can be.

For FPS, the gameplay is always the same (people kill you and you kill people).
In HaxBall, changing the map changes everything.

Suppose you're a player of LongBounce 3v3, in which only 1 goal can be scored. You play 100 games and score 60 goals, which is crazy good.
However, since most modes work with a 3 goal basis, your stats are gonna look poor, when in fact, with these stats you're probably the best LongBounce player.

And I'm not even gonna mention racing.

Stats are most effective when there's a minimum of structure.
No official/certified HaxBall rooms -> No global stats.

**Comment by AnddyAnddy** [comment]
I agree with what you say so I think that stats are simply personal nostalgy and can't be comparable, if there is no kind of ranking, players (except retards) won't have to farm, I'd love to see my global stats since 2011.

An idea behind this is being able to sort our stats as the player wants (a group of maps, a group of XvsX, enable/disable whenever we want).

I repeat again, since I don't think global stats can be reliable for rankings, imo, it just adds personal nostalgy for what we ever done in haxball.

**Comment by Raamyy** [comment]
I think one of the way to achieve this is to introduce like pro accounts for the game, which costs money.

This might prove that he is not messing up with the stats and if any stats spam detected his account should be disabled.

But anyways I'm not a fan with this option.

**Comment by AnddyAnddy** [comment]
You both don't look like to understand that the stats i'm talking about are not revelant to real rankings, thus, if anyone messes it up, it's his own business, with the features I talked about, any not dumb player will probably enjoy to have this since it's just an addition and not a "remove headless" or whatever.

**Comment by gamingmastar** [comment]
http://haxball.gq , check this goto statistics and see ranking

**Comment by AnddyAnddy** [comment]
This is restricted to some rooms, that's not what i'm talking about here
(nice site btw)
